Tools/clinic/clinic.py

```diff
@@ -2212,11 +2212,10 @@ def print_block(
 
                 """)
 
-            if clinic is not None:
```

Comment:
Hm, why didn't mypy flag this? This check should be superfluous in the current code.

Reply:
`¯\_(ツ)_/¯`

Reply:
I think as the annotation was ``clinic: Clinic | None`` -- signalling that ``None`` is a legal value. If we want to tighten to require passing these attributes that might make more sense?

Reply:
> I think as the annotation was `clinic: Clinic | None`

No, the current annotation is `clinic: Clinic`:

https://github.com/python/cpython/blob/cf7ba83eb274df8389cb9ebdf8601132c47de48a/Tools/clinic/clinic.py#L2175

Thinking about it, though, this perhaps doesn't fit in neatly to any of mypy's existing checks. It wouldn't flag a reachability warning, since the block is very-much reachable (the issue is the opposite: the condition is never _`False`_). And it wouldn't flag a `redundant-expr` statement either, since it's more of a `redundant-if-statement` than a `redundant-expr` situation.

So this is probably a case of a "missing mypy feature" rather than a "mypy bug" :)